### PR TITLE
updated Handlebars Generator

### PR DESCRIPTION
**Pull request by fokusferit**
There is an issue if you try to initiate the kssHandlebarsGenerator via the JS-API directly. If no custom helpers are defined the script is throwing an error because in line 109 the check this.config.helpers.length > 0 will throw an error if this.config.helpers = undefined .
This small change should fix it.

### CODE REVIEW

File: generator/handlebars/kss_handlebars_generator.js [hunk 1]
```diff
@@ -270,15 +272,17 @@ kssHandlebarsGenerator.generatePage = function(styleguide, sections, root, secti
...
         try {
-          files = glob.sync(this.config.source[key] + '/**/' + this.config.homepage);
+          files = glob.sync(this.config.homepage);
           if (files.length) {
             homepageText = ' ' + marked(fs.readFileSync(files[0], 'utf8'));
+            console.log('... parsing content from ' + this.config.homepage);
           }
         } catch (e) {
           // empty
+          console.log('There was an error reading the content in' + this.config.homepage);
```
Comment (JohnAlbin) [inline generator/handlebars/kss_handlebars_generator.js]: This change looks unrelated to the problem you describe.

Also, it's wrong. We need to search for the homepage in each of the source directories given since there can be multiple directories specified on the command line, e.g. `--source dir1 --source dir2`

File: generator/handlebars/kss_handlebars_generator.js [hunk 2]
```diff
@@ -302,13 +306,15 @@ kssHandlebarsGenerator.generatePage = function(styleguide, sections, root, secti
...
+      cssPath = path.relative(this.config.destination, this.config.css[key]);
+      styles = styles + '<link rel="stylesheet" href="' + cssPath + '">\n';
     }
   }
   for (key in this.config.js) {
-    if (this.config.js.hasOwnProperty(key)) {
-      scripts = scripts + '<script src="' + this.config.js[key] + '"></script>\n';
+    if (typeof this.config.destination !== 'undefined') {
+      jsPath = path.relative(this.config.destination, this.config.js[key]);
+      scripts = scripts + '<script src="' + jsPath + '"></script>\n';
```
Comment (JohnAlbin) [inline generator/handlebars/kss_handlebars_generator.js]: The `--js` option takes URLs, not paths relative to the current directory. You are free to pass paths relative to the destination directory in the `--js` option, since relative paths are the same as relative URLs in that case. But you cannot change the code to no longer accept full URLs like https://cdnjs.cloudflare.com/ajax/libs/jquery/2.1.4/jquery.min.js

File: generator/handlebars/kss_handlebars_generator.js (hunk 1 above)
Comment (fokusferit) [inline generator/handlebars/kss_handlebars_generator.js]: Ok, I understand that point, but wouldn't this make the project structure a bit complicated? Just maintaining one homepage.md could be sufficient.
